xdsl/backend/riscv/lowering/convert_riscv_scf_to_riscv_cf.py

```diff
@@ -0,0 +1,166 @@
+from xdsl.dialects import builtin, riscv, riscv_cf, riscv_scf
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class LowerRiscvScfForPattern(RewritePattern):
+    """
+    Create a CFG subgraph for the loop around its body blocks (if the body
+    contained other loops, they have been already lowered to a flow of blocks).
+    Maintain the invariants that a CFG subgraph created for any loop has a single
+    entry and a single exit, and that the entry/exit blocks are respectively
```

Comment:
I'm a bit confused here, because from what I understand we do not do anything specifically to maintain the single entry-exit invariant, right? we get this for free coming from `scf`.
The latter (i.e., placement of blocks) is fine.